sim/core/consumes.go

```diff
@@ -71,104 +82,198 @@ func ConsumesStats(c proto.Consumes) stats.Stats {
 	return s
 }
 
-func TryActivateDrums(sim *Simulation, character *Character) {
-	if character.IsOnCD(MagicIDDrums, sim.CurrentTime) {
-		return
-	}
+// Adds drums as a major cooldown to the character, if it's being used.
+func registerDrumsCD(agent Agent, consumes proto.Consumes) {
+	character := agent.GetCharacter()
+	drumsType := proto.Drums_DrumsUnknown
 
-	partyCast := character.Party.Buffs.Drums
-	if partyCast == proto.Drums_DrumsUnknown {
-		return
+	// Whether this agent is the one casting the drums.
+	//drumsSelfCast := false
+
+	if consumes.Drums != proto.Drums_DrumsUnknown {
+		drumsType = consumes.Drums
+		//drumsSelfCast = true
+	} else if character.Party.buffs.Drums != proto.Drums_DrumsUnknown {
+		drumsType = character.Party.buffs.Drums
```

Comment:
I assume this would need to be moved out when doing a 'raid' sim instead of an 'individual sim'